app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -440,6 +445,27 @@ public void onReceivedError(WebView view, int errorCode, String description, Str
         });
     }
 
+    private void setPageProtection() {
+        bridge.evaluate(JavaScriptActionHandler.getProtection(), value -> {
+            Protection protection = GsonUtil.getDefaultGson().fromJson(value, Protection.class);
+            if (model.getPage() != null) {
+                model.getPage().getPageProperties().setProtection(protection);
+                bridge.execute(JavaScriptActionHandler.setUpEditButtons(true, !model.getPage().getPageProperties().canEdit()));
```

Comment:
Do we actually need to execute `setUpEditButtons()` again? Since we already send the user's edit roles to the page in the `setup` callback, the page should be already aware if the user can edit the page.

Reply:
Correct!